core/types/types.cc

```diff
@@ -272,8 +281,13 @@ bool Types::canBeFalsy(const GlobalState &gs, const TypePtr &what) {
 TypePtr Types::approximateSubtract(const GlobalState &gs, const TypePtr &from, const TypePtr &what) {
     TypePtr result;
     typecase(
-        what, [&](const ClassType &c) { result = Types::dropSubtypesOf(gs, from, c.symbol); },
-        [&](const AppliedType &c) { result = Types::dropSubtypesOf(gs, from, c.klass); },
+        what,
+        [&](const ClassType &c) {
+            result = Types::dropSubtypesOf(gs, from, InlinedVector<ClassOrModuleRef, 1>{c.symbol});
+        },
+        [&](const AppliedType &c) {
+            result = Types::dropSubtypesOf(gs, from, InlinedVector<ClassOrModuleRef, 1>{c.klass});
```

Comment:
Can we construct spans directly here rather than indirecting through `InlinedVector`?